scripts/dnd5e-helpers.js

```diff
@@ -348,32 +359,112 @@ function includes_array(arr, comp) {
-      content: "<i>remains calm as a level " + spellLevel + " spell is cast...</i> ([[/r " + d20result + " #1d20 result]])",
-      speaker: ChatMessage.getSpeaker({ alias: "The Weave" })
-    });
+
+  if (d20result === 1) {
+    ShowSurgeResult('surges', spellLevel, `([[/r ${d20result} #1d20 result]])`);
+    RollOnWildTable(rollType);
+  } else {
+    ShowSurgeResult('remains calm', spellLevel, `([[/r ${d20result} #1d20 result]])`);
+  }
+}
+
+/** Role a more surge, checking against spell level cast */
+function RollForMoreSurge(spellLevel, rollType) {
+  const roll = new Roll("1d20").roll();
+  const d20result = +roll["result"];
+
+  if (game.settings.get('dnd5e-helpers', 'debug')) {
+    console.log(`DnD5e Helper: More Surge - ${d20result} vs ${spellLevel}`);
+  }
+
+  if (d20result <= spellLevel) {
+    ShowSurgeResult('surges', spellLevel, `([[/r ${d20result} #1d20 result]])`);
+    RollOnWildTable(rollType);
+  } else {
+    ShowSurgeResult('remains calm', spellLevel, `([[/r ${d20result} #1d20 result]])`);
+  }
+}
+
+/** Role a volatile surge, checking against spell level cast + d4 */
+function RollForVolatileSurge(spellLevel, rollType, actor) {
+  const wmToCFeatureName = (game.settings.get('dnd5e-helpers', 'wmToCFeatureName') !== '') 
+    ? game.settings.get('dnd5e-helpers', 'wmToCFeatureName') : wmToCFeatureDefault;
+  if (wmToCFeatureName !== '') {
+    /** if tides of chaos hasn't been used then no volatile roll */
+    if (!TidesOfChaosSpent(actor, wmToCFeatureName)) {
+      /** Tide of Chaos hasn't been used or has already reset, do more surge */
+      RollForMoreSurge(spellLevel, rollType);
+    } else {
+      const d20roll = new Roll("1d20").roll();
+      const d4roll = new Roll("1d4").roll();
+      const d20result = +d20roll["result"];
+      const d4result = +d4roll["result"];
+
+      if (game.settings.get('dnd5e-helpers', 'debug')) {
+        console.log(`DnD5e Helper: Volatile Surge - ${d20result} vs ${spellLevel + d4result}(Spell: ${spellLevel})(d4: ${d4result})`);
+      }
+
+      if (d20result <= (spellLevel + d4result)) {
+        ShowSurgeResult('surges', spellLevel, `([[/r ${d20result} #1d20 result]])`, `([[/r ${d4result} #1d4 result]])`);
+        RollOnWildTable(rollType);
+      } else {
+        ShowSurgeResult('remains calm', spellLevel, `([[/r ${d20result} #1d20 result]])`, `([[/r ${d4result}  #1d4 result]])`);
+      }
+    }
+  } else {
+    if (game.settings.get('dnd5e-helpers', 'debug')) {
+      console.log('Dnd5e Helper: Tides of Chaos feature name not setup');
+    }
   }
 }
 
+/** show surge result in chat box */
+function ShowSurgeResult(action, spellLevel, resultText, extraText = '') {
+  ChatMessage.create({
+    content: `<i>${action} as a level ${spellLevel} ${extraText} spell is cast!</i> ${resultText}`,
+    speaker: ChatMessage.getSpeaker({ alias: "The Weave" })
+  });
+}
+
+/** is the tides of chaos feature used */
+function TidesOfChaosSpent(actor, wmToCFeatureName) {
```

Comment:
I'm a little confused here regarding the tides of chaos feature. I was assuming the Tides of Chaos feature itself would have a number of uses and we reset the item's usage.

Here it looks like you are assuming the feature uses are stored in a resource?

I think this is the only thing that needs a rework now. Via a call like:
```suggestion
let tcItem = actor.items.getName(wmToCFeatureName);
await tcItem.update( /** reset charges */)
```

TidesOfCahosSpend will need to be async and return the promise given by item.update().